cmd/antrea-agent/options.go

```diff
@@ -146,4 +151,74 @@ func (o *Options) setDefaults() {
 	if o.config.APIPort == 0 {
 		o.config.APIPort = apis.AntreaAgentAPIPort
 	}
+
+	if o.config.FlowCollectorAddr != "" && o.config.PollAndExportInterval == "" {
+		o.pollingInterval = 5 * time.Second
+		o.exportInterval = 60 * time.Second
+	}
+}
+
+func (o *Options) validateFlowExporterConfig() error {
+	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
+		if o.config.FlowCollectorAddr != "" {
+			// Check if it is TCP or UDP
+			strSlice := strings.Split(o.config.FlowCollectorAddr, ":")
+			var proto string
+			if len(strSlice) == 2 {
+				// No separator "." and proto is given
+				proto = "tcp"
+			} else if len(strSlice) > 2 {
+				if strSlice[2] == "udp" {
+					proto = "udp"
+				} else {
+					// All other cases default proto is tcp
+					proto = "tcp"
+				}
+			} else {
+				return fmt.Errorf("IPFIX flow collector is given in invalid format")
+			}
+			// Convert the string input in net.Addr format
+			hostPortAddr := strSlice[0] + ":" + strSlice[1]
+			_, _, err := net.SplitHostPort(hostPortAddr)
+			if err != nil {
+				return fmt.Errorf("IPFIX flow collector is given in invalid format: %v", err)
+			}
+			if proto == "udp" {
+				o.flowCollector, err = net.ResolveUDPAddr("udp", hostPortAddr)
+				if err != nil {
+					return fmt.Errorf("IPFIX flow collector over UDP proto is not resolved: %v", err)
+				}
+			} else {
+				o.flowCollector, err = net.ResolveTCPAddr("tcp", hostPortAddr)
+				if err != nil {
+					return fmt.Errorf("IPFIX flow collector server TCP proto is not resolved: %v", err)
```

Comment:
server? why is the message not consistent with the udp one?

Reply:
Done.